core/src/main/java/org/elasticsearch/index/mapper/IdFieldMapper.java

```diff
@@ -113,32 +105,64 @@ public boolean isSearchable() {
 
         @Override
         public Query termQuery(Object value, @Nullable QueryShardContext context) {
-            final BytesRef[] uids = Uid.createUidsForTypesAndId(context.queryTypes(), value);
-            return new TermInSetQuery(UidFieldMapper.NAME, uids);
+            return termsQuery(Arrays.asList(value), context);
         }
 
         @Override
-        public Query termsQuery(List values, @Nullable QueryShardContext context) {
+        public Query termsQuery(List<?> values, @Nullable QueryShardContext context) {
+            if (indexOptions() != IndexOptions.NONE) {
+                return super.termsQuery(values, context);
+            }
             return new TermInSetQuery(UidFieldMapper.NAME, Uid.createUidsForTypesAndIds(context.queryTypes(), values));
```

Comment:
I might miss something but we are on the ID field that doesn't have a type? why are we using `Uid.createUidsForTypesAndIds(context.queryTypes(), values)`?